src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -2008,6 +2008,43 @@ class spell_q14386_call_attack_mastiffs : public SpellScript
         OnEffectHit += SpellEffectFn(spell_q14386_call_attack_mastiffs::HandleEffect, EFFECT_1, SPELL_EFFECT_SEND_EVENT);
     }
 };
+
+class spell_q59928_spell_ship_crash_teleport : public SpellScript
+{
+    PrepareSpellScript(spell_q59928_spell_ship_crash_teleport);
+
+    void RelocateTransportOffset(SpellEffIndex /*effIndex*/)
+    {
+        Unit* target = GetHitUnit();
+
+        if (TransportBase* transport = target->GetTransport())
+            transport->RemovePassenger(target);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_q59928_spell_ship_crash_teleport::RelocateTransportOffset, EFFECT_3, SPELL_EFFECT_TELEPORT_UNITS);
+    }
+};
+
+class spell_q58208_spell_ship_crash_teleport : public SpellScript
+{
+    PrepareSpellScript(spell_q58208_spell_ship_crash_teleport);
+
+    void RelocateTransportOffset(SpellEffIndex /*effIndex*/)
+    {
+        Unit* target = GetHitUnit();
+
+        if (TransportBase* transport = target->GetTransport())
+            transport->RemovePassenger(target);
```

Comment:
Do we know of any teleportspells which happen on the same transport (aka the exact opposite)? If not i'd suggest doing this implicitly for all teleport spells (aka in spell system, not in a script).
If such spells exist we might have to look at spellattributes or make some generic script

Dont be triggered by this malcrom, this is just missing info (maybe someone who read this has an idea) to prevent such code being duplicated 1000x times.

Reply:
I think this should be checked in Spell.cpp. I didn't move it to the exile's reach file because I'm not sure how TC wants to handle it.

Reply:
spell_quest is for generic quest spell scripts (used in multiple zones)
At least that was the plan

Reply:
Spells were scripted before I added the exiles reach file and still not sure how TC wants to handle this so not changing anything yet

Reply:
Modox I only get pissed of when you people come at my work, as my daughter said, like a bunch of gremlins.